UnityProject/Assets/Scripts/HealthV2/Living/BodyPartFunctionality.cs

```diff
@@ -1,24 +1,53 @@
 using System;
 using System.Collections.Generic;
 using UnityEngine;
+using Systems.Explosions;
+using NaughtyAttributes;
 
 namespace HealthV2
 {
-	public class BodyPartFunctionality : MonoBehaviour
+	public class BodyPartFunctionality : MonoBehaviour, IEmpAble
 	{
 		protected BodyPart bodyPart;
 		[NonSerialized]
 		public BodyPart RelatedPart;
 
+		public bool isEMPVunerable = false;
+
+		[ShowIf("isEMPVunerable")]
+		public int EMPResistance = 2;
+
 		public virtual void ImplantPeriodicUpdate(){}
 		public virtual void RemovedFromBody(LivingHealthMasterBase livingHealth){}
 		public virtual void AddedToBody(LivingHealthMasterBase livingHealth){}
 		public virtual void SetUpSystems(){}
 		public virtual void InternalDamageLogic() {}
+		
+		public virtual void OnEmp(int strength)
+		{
+			if (isEMPVunerable == false) return;
+
+			if(EMPResistance == 0)
```

Comment:
I think the code would be clearer if this check was merged with the `if` below. Something like `EMPResistance == 0 || DMMath.Prob(1 / EMPResistance)`.